diff --git a/cJSON.c b/cJSON.c
index 524ba46..d31731f 100644
--- a/cJSON.c
+++ b/cJSON.c
@@ -548,9 +548,8 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
     double d = item->valuedouble;
     int length = 0;
     size_t i = 0;
-    unsigned char number_buffer[26] = {0}; /* temporary buffer to print the number into */
+    unsigned char number_buffer[50] = {0}; /* temporary buffer to print the number into */
     unsigned char decimal_point = get_decimal_point();
-    double test = 0.0;
 
     if (output_buffer == NULL)
     {
@@ -568,6 +567,7 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
 	}
     else
     {
+#if 0
         /* Try 15 decimal places of precision to avoid nonsignificant nonzero digits */
         length = sprintf((char*)number_buffer, "%1.15g", d);
 
@@ -577,6 +577,8 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
             /* If not, print with 17 decimal places of precision */
             length = sprintf((char*)number_buffer, "%1.17g", d);
         }
+#endif
+        length = snprintf((char*)number_buffer, sizeof(number_buffer), "%.20g", d);
     }
 
     /* sprintf failed or buffer overrun occurred */
